distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/AbstractIntegrationTest.java

```diff
@@ -81,38 +82,109 @@
+    protected static String ddfHome;
 
-    protected static final String SERVICE_ROOT = SECURE_ROOT + HTTPS_PORT + "/services";
+    public enum Port {
+        BASE_PORT("org.codice.ddf.system.basePort"), HTTP_PORT(
+                "org.codice.ddf.system.httpPort"), HTTPS_PORT(
+                "org.codice.ddf.system.httpPort"), SSH_PORT, RMI_SERVER_PORT, RMI_REG_PORT;
 
-    protected static final String INSECURE_ROOT = "http://localhost:";
+        private final String systemProperty;
 
-    protected static final String INSECURE_SERVICE_ROOT = INSECURE_ROOT + HTTP_PORT + "/services";
+        Port() {
+            this.systemProperty = null;
+        }
 
-    protected static final String REST_PATH = SERVICE_ROOT + "/catalog/";
+        Port(String systemProperty) {
+            this.systemProperty = systemProperty;
+        }
 
-    protected static final String OPENSEARCH_PATH = REST_PATH + "query";
+        String getSystemProperty() {
+            return this.systemProperty;
+        }
 
-    protected static final String CSW_PATH = SERVICE_ROOT + "/csw";
+        String getPort(Integer basePort) {
+            return String.valueOf(basePort + this.ordinal());
+        }
 
-    protected static final String OPENSEARCH_SOURCE_ID = "openSearchSource";
+        String getPort() {
+            return String.valueOf(basePort + this.ordinal());
+        }
 
-    protected static final String CSW_SOURCE_ID = "cswSource";
+        String getUrl(Integer basePort) {
+            return this.getPort(basePort);
+        }
 
-    protected static final String DDF_HOME_PROPERTY = "ddf.home";
+    }
 
-    protected static String ddfHome;
+    public enum Url {
+        SECURE_ROOT("https://localhost:"),
+        SERVICE_ROOT(SECURE_ROOT, Port.HTTPS_PORT, "/services"),
+        INSECURE_ROOT("http://localhost:"),
+        INSECURE_SERVICE_ROOT(INSECURE_ROOT, Port.HTTP_PORT, "/services"),
+        REST_PATH(SERVICE_ROOT, "/catalog/"),
+        OPENSEARCH_PATH(REST_PATH, "query"),
+        CSW_PATH(SERVICE_ROOT, "/csw"),
+        ADMIN_ALL_SOURCES_PATH(SECURE_ROOT, Port.HTTPS_PORT,
+                "/jolokia/exec/org.codice.ddf.catalog.admin.plugin.AdminSourcePollerServiceBean:service=admin-source-poller-service/allSourceInfo"),
+        ADMIN_STATUS_PATH(SECURE_ROOT, Port.HTTPS_PORT,
+                "/jolokia/exec/org.codice.ddf.catalog.admin.plugin.AdminSourcePollerServiceBean:service=admin-source-poller-service/sourceStatus/");
+
+        private final String root;
+
+        private final Enum[] pieces = new Enum[2];
+
+        Url(String base) {
+            this.root = base;
+        }
+
+        Url(Enum piece, String base) {
+            this.pieces[0] = piece;
+            this.root = base;
+        }
+
+        Url(Enum piece0, Enum piece1, String base) {
+            this.pieces[0] = piece0;
+            this.pieces[1] = piece1;
+            this.root = base;
+        }
+
+        String getUrl(Integer basePort) {
```

Comment:
If you change your `Enum[]` to a `List<Enum>`, you should be able to replace this method's guts with...
```
            return pieces.stream()
                    .map(p -> {
                        if (p instanceof Port) {
                            return ((Port) p).getPort(basePort);
                        } else if (p instanceof Url) {
                            return ((Url) p).getUrl(basePort);
                        }
                        return "";
                    })
                    .reduce("", (acc, val) -> acc + val) + root;
```